src/forward.js

```diff
@@ -0,0 +1,111 @@
+const http = require('http');
+const https = require('https');
+const stream = require('stream');
+const util = require('util');
+const { isHopByHopHeader } = require('./tools');
+
+const pipeline = util.promisify(stream.pipeline);
+
+const validHeadersOnly = (array) => {
+    const rawHeaders = [];
+
+    let containsHost = false;
+
+    for (let i = 0; i < array.length; i += 2) {
+        const name = array[i];
+        const value = array[i + 1];
+
+        try {
+            http.validateHeaderName(name);
+            http.validateHeaderValue(name, value);
+        } catch (error) {
+            continue;
+        }
+
+        if (isHopByHopHeader(name)) {
+            continue;
+        }
+
+        if (name.toLowerCase() === 'host') {
+            if (containsHost) {
+                continue;
+            }
+
+            containsHost = true;
+        }
+
+        rawHeaders.push(name, value);
+    }
+
+    return rawHeaders;
+};
+
+// eslint-disable-next-line no-async-promise-executor
+const forward = async (request, response, handlerOpts) => new Promise(async (resolve, reject) => {
```

Comment:
Please can you add comments here please what the function does, and what are the params? Perhaps also describe why this is not inherited from HandlerBase, it's not immediately obvious.

Reply:
> Perhaps also describe why this is not inherited from HandlerBase, it's not immediately obvious.

My idea is that each handler will be a function like this, there's no need for `HandlerBase` at all.

Reply:
Yeah, just pls comment what and why :)